src/lib/broadcast.ts

```diff
@@ -1,5 +1,26 @@
 import { hasOpener } from "./has-opener";
 
+// this method is mostly to reduce the number of console.error messages
+// when the origin is set to anything other than the default (*)
+function shouldBroadcast(frameToBroadcastTo: Window, origin: string): boolean {
+  // this is the default origin, so we can short circuit any logic around this
+  // if the origin is * and just tell it to broadcast
+  if (origin === "*") {
+    return true;
+  }
+
+  // this is wrapped in a try catch because if the domain is different than
+  // the domain of the frame issuing the postMessage, we will not be able
+  // to access the origin attribute and it will throw instead
+  // in that case, we _must_ broadcast to it even if the origin does not match,
+  // resulting in an ugly (but benign) console.error
+  try {
+    return frameToBroadcastTo.origin === origin;
```

Comment:
Does this need to be `frameToBroadcastTo.location.origin`?

Reply:
Looks like it doesn't "need" to be, but I think using `location` would allow this to work on older browsers.

Reply:
So there may be some subtle differences between the two. For example:
```js
var frame = document.createElement("iframe")
frame.onload = function() {
  var frameWin = frame.contentWindow
  console.log(frameWin.location.origin) // "null"
  console.log(frameWin.origin) // "https://stargate.example"
}
document.body.appendChild(frame)
```
From https://html.spec.whatwg.org/multipage/webappapis.html#windoworworkerglobalscope-mixin